lib/nexmo_jwt/jwt_builder.rb

```diff
@@ -44,24 +46,24 @@ class JWTBuilder
     def initialize(params = {})
       Nexmo::JWTBuilder.validate_parameters_not_conflicting(params)
 
-      @application_id = set_application_id(params.fetch(:application_id))
-      @private_key = set_private_key(params.fetch(:private_key))
+      @application_id = application_id!(params.fetch(:application_id))
+      @private_key = private_key!(params.fetch(:private_key))
       @jti = params.fetch(:jti, SecureRandom.uuid)
       @nbf = params.fetch(:nbf, nil)
       @ttl = params.fetch(:ttl, 900)
       @exp = params.fetch(:exp, nil)
       @alg = params.fetch(:alg, 'RS256')
       @paths = params.fetch(:paths, nil)
       @subject = params.fetch(:subject, 'Subject')
-      @jwt = Nexmo::JWT.new(:generator => self)
+      @jwt = Nexmo::JWT.new(generator: self)
 
       after_initialize!(self)
     end
 
     def self.validate_parameters_not_conflicting(params)
-      if params[:ttl] && params[:exp]
-        raise ArgumentError, "Expected either 'ttl' or 'exp' parameter, preference is to set 'ttl' parameter"
-      end
+      return unless params[:ttl] && params[:exp]
```

Comment:
Please revert the `if` statement, would be happy to consider this in a separate PR.